src/decode.rs

```diff
@@ -4587,172 +4587,42 @@ unsafe fn decode_b(
-                            if t.warpmv.matrix[4 as libc::c_int as usize]
-                                < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.matrix[4 as libc::c_int as usize]),
-                            if t.warpmv.matrix[5 as libc::c_int as usize]
-                                < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.matrix[5 as libc::c_int as usize]),
-                            if (t.warpmv.u.p.alpha as libc::c_int) < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.alpha as libc::c_int),
-                            if (t.warpmv.u.p.beta as libc::c_int) < 0 as libc::c_int {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.beta as libc::c_int),
-                            if (t.warpmv.u.p.gamma as libc::c_int) < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.gamma as libc::c_int),
-                            if (t.warpmv.u.p.delta as libc::c_int) < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.delta as libc::c_int),
-                            b
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .mv2d
-                                .c2rust_unnamed
-                                .y as libc::c_int,
-                            b
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .mv2d
-                                .c2rust_unnamed
-                                .x as libc::c_int,
+                        println!(
+                            "[ {:+x} {:+x} {:+x}\n  {:+x} {:+x} {:+x} ]\n\
```

Comment:
The `signabs` macro expanded above is slightly different from this.  It prints `-` if negative or else ` `, not `+`, which I believe is what the `+` sign formatting flag does in Rust.  I think we should just deduplicate the `signabs` macro.

Reply:
What does the `x` in `{:+x}` mean?

Reply:
The `x` in `{:+x}` means that the value is formatted as a hexadecimal number.

You're right, I completely mistranslated what this print output was doing (I entirely forgot the `abs` invocation!). I've changed it to a new version that translates `signabs` into a Rust helper struct, it should behave closer to the original now.

Reply:
a0c31084bb60ab3564603136ef06bc1f18eb0a71 looks good.